elastic/kibana
# [Synthetics] ensure metric color and current status are synced

## Summary

Relates to https://github.com/elastic/kibana/issues/146001
Resolves https://github.com/elastic/kibana/issues/146565

This PR ensures that the status of monitors is in sync with the current status metric by deriving the status of each individual monitor card based on the current status api.

Updates the current status api to [return objects representing all the down monitors and all the up monitors](https://github.com/elastic/kibana/pull/145031/files#diff-db953498cb683d4f81a105a8c53f80ef08aea4bb687e6a29f651dd4e148bcc55R122), with the keys being the combination of the monitor config id and location name.

These two objects are [combined into one object in redux](https://github.com/elastic/kibana/pull/145031/files#diff-2fb9b43772c61e2706a828651abd651a2a03b33b9f7b59c91d91c3a245edf655R77). 

Creates a new `useStatusByLocationOverview` [hook](https://github.com/elastic/kibana/pull/145031/files#diff-b80b952c8c3e15be0472d8b63a23f6ca0798dfbfdbce6a1ed056d7f954ab200eR11) that consumes this object. 

[Uses that hook](https://github.com/elastic/kibana/pull/145031/files#diff-d65b7f3fa061b4f65d63f72296d29f603e10757e595dd688a34fa99deeac3250R56) in Overview Metric item to derive the status. 

### Testing

1. Create a handful of monitors, ensuring you have at least 1 ui monitor and 1 project monitor
2. Ensure your cluster is connected to the synthetics service, either by running the service locally or connecting to dev environment
3. Navigate to overview. Ensure the number of down monitor cards and number of up monitor cards matches the current status metric
4. Use the sort by status feature to test for regression. That logic was lightly touched in this PR
